From: Matthias Bach <marix@marix.org>
Date: 2019-01-13T20:19:23+0100
Subject: Fix compilation of the TinyXML parser.
References: none
Upstream: no, backported from upstream

This is a backport of a fix that upstream only added after release in commit 0a916f5.
diff -rub cegui-0.8.7o/cegui/src/XMLParserModules/TinyXML/XMLParser.cpp cegui-0.8.7/cegui/src/XMLParserModules/TinyXML/XMLParser.cpp
--- cegui-0.8.7o/cegui/src/XMLParserModules/TinyXML/XMLParser.cpp	2016-04-23 20:19:40.000000000 +0200
+++ cegui-0.8.7/cegui/src/XMLParserModules/TinyXML/XMLParser.cpp	2019-01-13 20:59:19.490972458 +0100
@@ -130,7 +130,7 @@
                 processElement(childNode->ToElement());
                 break;
             case TiXmlNode::CEGUI_TINYXML_TEXT:
-                if (childNode->ToText()->Value() != '\0')
+                if (childNode->ToText()->Value() != 0)
                     d_handler->text((encoded_char*)childNode->ToText()->Value());
                 break;
 
